src/playlist-merge.js

```diff
@@ -0,0 +1,583 @@
+  const oldSegments = oldPlaylist.segments || [];
+  const newSegments = newPlaylist.segments || [];
+  // At the time of writing, mpd-parser did not add discontinuitySequence. This code
+  // allows for it to exist or to not exist, and should handle both appropriately.
+  const oldDiscontinuitySequence = oldPlaylist.discontinuitySequence || 0;
+
+  // If the prior playlist had no segments, it's likely either the start of the stream or
+  // a break in the stream. Either way, the best approach is to consider the new segments
+  // a discontinuous region of the media and to continue as if the new playlist is a
+  // continuation of the prior stream.
+  if (oldSegments.length === 0) {
+    newPlaylist.mediaSequence = oldPlaylist.mediaSequence;
+    // The discontinuity sequence should remain the same. Accounting for removed
+    // discontinuities occurs on playlist updates where segments are removed.
+    newPlaylist.discontinuitySequence = oldDiscontinuitySequence;
+    // The timeline should remain the same until a new segment is added.
+    newPlaylist.timeline = oldPlaylist.timeline + (newPlaylist.segments.length ? 1 : 0);
+
+    if (newSegments.length > 0) {
+      newSegments[0].discontinuity = true;
+    }
+
+    repositionSegmentsOnTimeline({
+      segments: newSegments,
+      mediaSequenceStart: newPlaylist.mediaSequence,
+      timelineStart: newPlaylist.timeline
+    });
+
+    newPlaylist.discontinuityStarts = findIndexes(newSegments, 'discontinuity');
+    return;
+  }
+
+  // If there are no new segments then the stream is either reaching an end or a gap.
+  // The media sequence should be updated accordingly, but the timeline should remain the
+  // same, as it will be updated once segments are added.
+  if (newSegments.length === 0) {
+    newPlaylist.mediaSequence = oldPlaylist.mediaSequence + oldSegments.length;
+    newPlaylist.discontinuitySequence =
+      oldDiscontinuitySequence + findIndexes(oldSegments, 'discontinuity').length;
+    newPlaylist.timeline = oldSegments.length ?
+      oldSegments[oldSegments.length - 1].timeline : oldPlaylist.timeline;
+    newPlaylist.discontinuityStarts = [];
+    return;
+  }
+
+  // From here on both old and new playlist have segments.
+  //
+  // To update the new playlist, we must position the old and the new together. The
+  // simplest approach is to try to match an exact segment and adjust timing values from
+  // there. However, this will not always work. It's possible that different periods will
+  // reuse the same segment, and it's also possible that refreshes may change certain
+  // segment properties (e.g., the URI).
+  //
+  // Because of these limitations, the best approach for matching segments is by finding
+  // a segment with a matching presentation time. These values should not change on
+  // refreshes.
+  const firstNewSegment = newSegments[0];
+  const oldMatchingSegmentIndex = oldSegments.findIndex((oldSegment) =>
```

Comment:
`findIndex` isn't available on IE11, unfortunately. And we should avoid breaking it if we can.